benchmarks/user/benchmark.py

```diff
@@ -177,8 +178,13 @@ def cli_bench(problem, **kwargs):
     """
     Complete benchmark with multiple simulation and performance parameters.
     """
+
     mode_benchmark()
-    kwargs['autotune'] = configuration['autotuning'].level
+    if kwargs['autotune'] != configuration['autotuning'].level:
```

Comment:
why is this in `bench`, but not in `run`?

probably this should be a callback when accessing the `autotune` parameter for the first time from click. See [here](https://github.com/opesci/devito/blob/master/benchmarks/user/benchmark.py#L45) for example -- it makes sure that if one runs with `-so 12`, then the argument value is interpreted for consistency as `[12]` (just like `-so 12 16` becomes [12, 16]) instead of `16` (ie, an int instead of a list)

so with a suitable callback you make it return a tuple such as `('aggressive', 'preemptive')` where clearly `'preemptive'` is fixed, while 'aggressive' depends on whatever the user has provided in input

Reply:
so if you do it this way, you automatically get the right and **same** thing in all modes

Reply:
I am a bit confused here...let me work it out...

Reply:
should it not be in both? why should `run` behave differently than `bench` when it gets to autotuning?

you have to understand how click callbacks work, I think. And you have an example right above, so it shouldn't be too difficult

Reply:
I am not contrary to that. I speak from the view of avoiding duplicate code.

Reply:
added, currently duplicated